Error creating bean with name 'amazonS3'

**Describe the bug**
Spring Content application is failing to start without valid S3 configuration, even though I don't want to use S3 for local development.

**To Reproduce**
1. Update to spring content 1.2.6
2. Add the starter "com.github.paulcwarren:spring-content-s3-boot-starter"
3. Do not provide valid configuration
4. Start the application

Exception: 
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
[ERROR] 2021-11-10 10:39:27.267 [restartedMain]  o.s.b.SpringApplication - Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'amazonS3' defined in class path resource [internal/org/springframework/content/s3/boot/autoconfigure/S3ContentAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.amazonaws.services.s3.AmazonS3]: Factory method 'amazonS3' threw exception; nested exception is com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in the builder or setup environment to supply a region.
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:658)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:638)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1352)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1195)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:944)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332)
	at eu.xenit.neutra.neutraarchive.NeutraArchiveApplication.main(NeutraArchiveApplication.java:30)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.amazonaws.services.s3.AmazonS3]: Factory method 'amazonS3' threw exception; nested exception is com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in the builder or setup environment to supply a region.
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
	... 24 common frames omitted
Caused by: com.amazonaws.SdkClientException: Unable to find a region via the region provider chain. Must provide an explicit region in the builder or setup environment to supply a region.
	at com.amazonaws.client.builder.AwsClientBuilder.setRegion(AwsClientBuilder.java:462)
	at com.amazonaws.client.builder.AwsClientBuilder.configureMutableProperties(AwsClientBuilder.java:424)
	at com.amazonaws.client.builder.AwsSyncClientBuilder.build(AwsSyncClientBuilder.java:46)
	at internal.org.springframework.content.s3.boot.autoconfigure.S3ContentAutoConfiguration.amazonS3(S3ContentAutoConfiguration.java:54)
	at internal.org.springframework.content.s3.boot.autoconfigure.S3ContentAutoConfiguration$$EnhancerBySpringCGLIB$$83ad0c1.CGLIB$amazonS3$0(<generated>)
	at internal.org.springframework.content.s3.boot.autoconfigure.S3ContentAutoConfiguration$$EnhancerBySpringCGLIB$$83ad0c1$$FastClassBySpringCGLIB$$2cc5f9a7.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331)
	at internal.org.springframework.content.s3.boot.autoconfigure.S3ContentAutoConfiguration$$EnhancerBySpringCGLIB$$83ad0c1.amazonS3(<generated>)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	... 25 common frames omitted

**Expected behavior**
Application starting correctly as the AmazonS3 bean is not used.

**Additional context**
On 1.2.5, the same application was working.

--- a/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/fs/boot/autoconfigure/FilesystemContentAutoConfiguration.java
+++ b/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/fs/boot/autoconfigure/FilesystemContentAutoConfiguration.java
@@ -8,6 +8,7 @@ import org.slf4j.LoggerFactory;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.content.fs.io.FileSystemResourceLoader;
 import org.springframework.context.annotation.Bean;
@@ -23,6 +24,11 @@ import internal.org.springframework.versions.jpa.boot.autoconfigure.JpaVersionsA
 @Configuration
 @AutoConfigureAfter({ JpaVersionsAutoConfiguration.class })
 @ConditionalOnClass(FilesystemStoreRegistrar.class)
+@ConditionalOnProperty(
+        prefix="spring.content.storage.type",
+        name = "default",
+        havingValue = "fs",
+        matchIfMissing=true)
 public class FilesystemContentAutoConfiguration {
 
 	@Configuration

--- a/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/jpa/boot/autoconfigure/JpaContentAutoConfiguration.java
+++ b/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/jpa/boot/autoconfigure/JpaContentAutoConfiguration.java
@@ -6,6 +6,7 @@ import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
@@ -21,6 +22,11 @@ import internal.org.springframework.versions.jpa.boot.autoconfigure.JpaVersionsA
 @Configuration
 @AutoConfigureAfter({DataSourceAutoConfiguration.class, JpaVersionsAutoConfiguration.class})
 @ConditionalOnClass({ DataSource.class, JpaStoresRegistrar.class })
+@ConditionalOnProperty(
+        prefix="spring.content.storage.type",
+        name = "default",
+        havingValue = "jpa",
+        matchIfMissing=true)
 @EnableConfigurationProperties(ContentJpaProperties.class)
 public class JpaContentAutoConfiguration {
 

--- a/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/mongo/boot/autoconfigure/MongoContentAutoConfiguration.java
+++ b/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/mongo/boot/autoconfigure/MongoContentAutoConfiguration.java
@@ -3,6 +3,7 @@ package internal.org.springframework.content.mongo.boot.autoconfigure;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
@@ -16,6 +17,11 @@ import internal.org.springframework.versions.jpa.boot.autoconfigure.JpaVersionsA
 @Configuration
 @AutoConfigureAfter({ JpaVersionsAutoConfiguration.class })
 @ConditionalOnClass({ MongoClient.class, MongoContentStoresRegistrar.class })
+@ConditionalOnProperty(
+        prefix="spring.content.storage.type",
+        name = "default",
+        havingValue = "gridfs",
+        matchIfMissing=true)
 @ConditionalOnMissingBean(MongoStoreFactoryBean.class)
 @Import({ MongoContentAutoConfigureRegistrar.class, MongoStoreConfiguration.class })
 public class MongoContentAutoConfiguration {

--- a/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/s3/boot/autoconfigure/S3ContentAutoConfiguration.java
+++ b/spring-content-autoconfigure/src/main/java/internal/org/springframework/content/s3/boot/autoconfigure/S3ContentAutoConfiguration.java
@@ -3,6 +3,7 @@ package internal.org.springframework.content.s3.boot.autoconfigure;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
@@ -25,6 +26,11 @@ import internal.org.springframework.versions.jpa.boot.autoconfigure.JpaVersionsA
 @Configuration
 @AutoConfigureAfter({ JpaVersionsAutoConfiguration.class })
 @ConditionalOnClass(AmazonS3Client.class)
+@ConditionalOnProperty(
+        prefix="spring.content.storage.type",
+        name = "default",
+        havingValue = "s3",
+        matchIfMissing=true)
 public class S3ContentAutoConfiguration {
 
 	@Configuration

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/fs/boot/autoconfigure/ContentFilesystemAutoConfigurationTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/fs/boot/autoconfigure/ContentFilesystemAutoConfigurationTest.java
@@ -1,4 +1,4 @@
-package org.springframework.content.fs.boot;
+package org.springframework.content.fs.boot.autoconfigure;
 
 import com.github.paulcwarren.ginkgo4j.Ginkgo4jConfiguration;
 import com.github.paulcwarren.ginkgo4j.Ginkgo4jRunner;

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/fs/boot/autoconfigure/FilesystemPropertiesTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/fs/boot/autoconfigure/FilesystemPropertiesTest.java
@@ -1,4 +1,4 @@
-package org.springframework.content.fs.boot;
+package org.springframework.content.fs.boot.autoconfigure;
 
 import org.junit.runner.RunWith;
 

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/fs/boot/defaultstorage/FilesystemAutoConfigurationTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/fs/boot/defaultstorage/FilesystemAutoConfigurationTest.java
@@ -1,0 +1,111 @@
+package org.springframework.content.fs.boot.defaultstorage;
+
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.AfterEach;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.BeforeEach;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.Context;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.Describe;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.It;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.mongo.MongoAutoConfiguration;
+import org.springframework.content.commons.repository.ContentStore;
+import org.springframework.content.fs.io.FileSystemResourceLoader;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.jmx.support.RegistrationPolicy;
+import org.springframework.support.TestEntity;
+
+import com.github.paulcwarren.ginkgo4j.Ginkgo4jConfiguration;
+import com.github.paulcwarren.ginkgo4j.Ginkgo4jRunner;
+
+import internal.org.springframework.content.elasticsearch.boot.autoconfigure.ElasticsearchAutoConfiguration;
+import internal.org.springframework.content.jpa.boot.autoconfigure.JpaContentAutoConfiguration;
+import internal.org.springframework.content.mongo.boot.autoconfigure.MongoContentAutoConfiguration;
+import internal.org.springframework.content.renditions.boot.autoconfigure.RenditionsContentAutoConfiguration;
+import internal.org.springframework.content.s3.boot.autoconfigure.S3ContentAutoConfiguration;
+import internal.org.springframework.versions.jpa.boot.autoconfigure.JpaVersionsAutoConfiguration;
+
+@RunWith(Ginkgo4jRunner.class)
+@Ginkgo4jConfiguration(threads = 1)
+public class FilesystemAutoConfigurationTest {
+
+	{
+		Describe("FilesystemContentAutoConfiguration", () -> {
+
+            Context("given a default storage type of fs", () -> {
+                BeforeEach(() -> {
+                    System.setProperty("spring.content.storage.type.default", "fs");
+                });
+                AfterEach(() -> {
+                    System.clearProperty("spring.content.storage.type.default");
+                });
+                It("should create an FileSystemResourceLoader bean", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfig.class);
+                    context.refresh();
+
+                    assertThat(context.getBean(FileSystemResourceLoader.class), is(not(nullValue())));
+                });
+            });
+
+            Context("given a default storage type other than fs", () -> {
+                BeforeEach(() -> {
+                    System.setProperty("spring.content.storage.type.default", "s3");
+                });
+                AfterEach(() -> {
+                    System.clearProperty("spring.content.storage.type.default");
+                });
+                It("should not create an FileSystemResourceLoader bean", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfig.class);
+                    context.refresh();
+
+                    try {
+                        context.getBean(FileSystemResourceLoader.class);
+                        fail("context should not have a a FileSystemResourceLoader bean");
+                    } catch (NoSuchBeanDefinitionException nsbe) {
+                    }
+                });
+            });
+
+            Context("given no default storage type", () -> {
+
+                It("should create an FileSystemResourceLoader bean", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfig.class);
+                    context.refresh();
+
+                    assertThat(context.getBean(FileSystemResourceLoader.class), is(not(nullValue())));
+                });
+            });
+        });
+	}
+
+	@Configuration
+    @EnableAutoConfiguration(exclude= {
+            ElasticsearchAutoConfiguration.class,
+            MongoAutoConfiguration.class,
+            JpaContentAutoConfiguration.class,
+            JpaVersionsAutoConfiguration.class,
+            MongoContentAutoConfiguration.class,
+            RenditionsContentAutoConfiguration.class,
+            S3ContentAutoConfiguration.class})
+	@EnableMBeanExport(registration = RegistrationPolicy.IGNORE_EXISTING)
+	public static class TestConfig {
+	}
+
+	public interface TestEntityRepository extends JpaRepository<TestEntity, Long> {
+	}
+
+	public interface TestEntityContentRepository extends ContentStore<TestEntity, String> {
+	}
+}

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/jpa/boot/autoconfigure/ContentJpaAutoConfigurationTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/jpa/boot/autoconfigure/ContentJpaAutoConfigurationTest.java
@@ -1,4 +1,4 @@
-package org.springframework.content.jpa.boot;
+package org.springframework.content.jpa.boot.autoconfigure;
 
 import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.AfterEach;
 import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.BeforeEach;

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/jpa/boot/autoconfigure/ContentJpaDatabaseInitializerTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/jpa/boot/autoconfigure/ContentJpaDatabaseInitializerTest.java
@@ -1,4 +1,4 @@
-package org.springframework.content.jpa.boot;
+package org.springframework.content.jpa.boot.autoconfigure;
 
 import com.github.paulcwarren.ginkgo4j.Ginkgo4jRunner;
 import internal.org.springframework.content.jpa.boot.autoconfigure.ContentJpaDatabaseInitializer;

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/s3/boot/autoconfigure/ContentS3AutoConfigurationTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/s3/boot/autoconfigure/ContentS3AutoConfigurationTest.java
@@ -1,4 +1,4 @@
-package org.springframework.content.s3.boot;
+package org.springframework.content.s3.boot.autoconfigure;
 
 import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.AfterEach;
 import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.BeforeEach;
@@ -13,7 +13,6 @@ import static org.mockito.Mockito.mock;
 
 import java.lang.reflect.Field;
 import java.net.URI;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
@@ -27,6 +26,7 @@ import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.support.TestEntity;
+import org.springframework.support.TestUtils;
 import org.springframework.util.ReflectionUtils;
 
 import com.amazonaws.AmazonWebServiceClient;
@@ -52,7 +52,7 @@ public class ContentS3AutoConfigurationTest {
 		    props.put("AWS_REGION", Regions.US_WEST_1.getName());
 		    props.put("AWS_ACCESS_KEY_ID", "user");
 		    props.put("AWS_SECRET_KEY", "password");
-		    setEnv(props);
+		    TestUtils.setEnv(props);
 		} catch (Exception e) {
 		    e.printStackTrace();
 		}
@@ -131,29 +131,28 @@ public class ContentS3AutoConfigurationTest {
                     System.clearProperty("spring.content.s3.secretKey");
                     System.clearProperty("spring.content.s3.pathStyleAccess");
                 });
-                It("should have a filesystem properties bean with the correct root set",
-                        () -> {
-                            AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
-                            context.register(TestConfigWithProperties.class);
-                            context.refresh();
+                It("should have a filesystem properties bean with the correct root set", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfigWithProperties.class);
+                    context.refresh();
 
-                            AmazonS3 client = context.getBean(AmazonS3.class);
+                    AmazonS3 client = context.getBean(AmazonS3.class);
 
-                            Field endpointField = getField(AmazonWebServiceClient.class, "endpoint");
-                            URI endpoint = (URI) endpointField.get(client);
-                            assertThat(endpoint.toString(), is("http://some-endpoint"));
+                    Field endpointField = getField(AmazonWebServiceClient.class, "endpoint");
+                    URI endpoint = (URI) endpointField.get(client);
+                    assertThat(endpoint.toString(), is("http://some-endpoint"));
 
-                            Field providerField = getField(AmazonS3Client.class, "awsCredentialsProvider");
-                            AWSCredentialsProvider provider = (AWSCredentialsProvider) providerField.get(client);
-                            assertThat(provider.getCredentials().getAWSAccessKeyId(), is("foo"));
-                            assertThat(provider.getCredentials().getAWSSecretKey(), is("bar"));
+                    Field providerField = getField(AmazonS3Client.class, "awsCredentialsProvider");
+                    AWSCredentialsProvider provider = (AWSCredentialsProvider) providerField.get(client);
+                    assertThat(provider.getCredentials().getAWSAccessKeyId(), is("foo"));
+                    assertThat(provider.getCredentials().getAWSSecretKey(), is("bar"));
 
-                            Field coField = getField(AmazonS3Client.class, "clientOptions");
-                            S3ClientOptions options = (S3ClientOptions) coField.get(client);
-                            assertThat(options.isPathStyleAccess(), is(true));
+                    Field coField = getField(AmazonS3Client.class, "clientOptions");
+                    S3ClientOptions options = (S3ClientOptions) coField.get(client);
+                    assertThat(options.isPathStyleAccess(), is(true));
 
-                            context.close();
-                        });
+                    context.close();
+                });
             });
 		});
 	}
@@ -202,32 +201,5 @@ public class ContentS3AutoConfigurationTest {
         Field f = ReflectionUtils.findField(clazz, field);
         ReflectionUtils.makeAccessible(f);
         return f;
-    }
-
-    private static void setEnv(Map<String, String> newenv) throws Exception {
-        try {
-            Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
-            Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
-            theEnvironmentField.setAccessible(true);
-            Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
-            env.putAll(newenv);
-            Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField("theCaseInsensitiveEnvironment");
-            theCaseInsensitiveEnvironmentField.setAccessible(true);
-            Map<String, String> cienv = (Map<String, String>)theCaseInsensitiveEnvironmentField.get(null);
-            cienv.putAll(newenv);
-        } catch (NoSuchFieldException e) {
-            Class[] classes = Collections.class.getDeclaredClasses();
-            Map<String, String> env = System.getenv();
-            for(Class cl : classes) {
-                if("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
-                    Field field = cl.getDeclaredField("m");
-                    field.setAccessible(true);
-                    Object obj = field.get(env);
-                    Map<String, String> map = (Map<String, String>) obj;
-                    map.clear();
-                    map.putAll(newenv);
-                }
-            }
-        }
     }
 }

--- a/spring-content-autoconfigure/src/test/java/org/springframework/content/s3/boot/defaultstorage/S3AutoConfigurationTest.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/content/s3/boot/defaultstorage/S3AutoConfigurationTest.java
@@ -1,0 +1,118 @@
+package org.springframework.content.s3.boot.defaultstorage;
+
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.AfterEach;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.BeforeEach;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.Context;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.Describe;
+import static com.github.paulcwarren.ginkgo4j.Ginkgo4jDSL.It;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.springframework.support.TestUtils.setEnv;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.content.commons.repository.ContentStore;
+import org.springframework.content.s3.config.EnableS3Stores;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.support.TestEntity;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.github.paulcwarren.ginkgo4j.Ginkgo4jConfiguration;
+import com.github.paulcwarren.ginkgo4j.Ginkgo4jRunner;
+
+@RunWith(Ginkgo4jRunner.class)
+@Ginkgo4jConfiguration(threads = 1)
+public class S3AutoConfigurationTest {
+
+	static {
+		mock(AmazonS3.class);
+
+		try {
+		    Map<String,String> props = new HashMap<>();
+		    props.put("AWS_REGION", Regions.US_WEST_1.getName());
+		    props.put("AWS_ACCESS_KEY_ID", "user");
+		    props.put("AWS_SECRET_KEY", "password");
+		    setEnv(props);
+		} catch (Exception e) {
+		    e.printStackTrace();
+		}
+	}
+
+	{
+		Describe("S3 auto configuration with default storage", () -> {
+
+            Context("given a default storage type of s3", () -> {
+                BeforeEach(() -> {
+                    System.setProperty("spring.content.storage.type.default", "s3");
+                });
+                AfterEach(() -> {
+                    System.clearProperty("spring.content.storage.type.default");
+                });
+                It("should create an AmazonS3 bean", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfigWithoutBeans.class);
+                    context.refresh();
+
+                    assertThat(context.getBean(AmazonS3.class), is(not(nullValue())));
+                });
+            });
+
+    		Context("given a default storage type other than s3", () -> {
+                BeforeEach(() -> {
+                    System.setProperty("spring.content.storage.type.default", "fs");
+                });
+                AfterEach(() -> {
+                    System.clearProperty("spring.content.storage.type.default");
+                });
+                It("should not create an AmazonS3 bean", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfigWithoutBeans.class);
+                    context.refresh();
+
+                    try {
+                        context.getBean(AmazonS3.class);
+                        fail("expected no AmazonS3 bean but bean found");
+                    } catch (NoSuchBeanDefinitionException nsbe) {
+                    }
+                });
+            });
+
+            Context("given no default storage type", () -> {
+
+                It("should create an AmazonS3 bean", () -> {
+                    AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+                    context.register(TestConfigWithoutBeans.class);
+                    context.refresh();
+
+                    assertThat(context.getBean(AmazonS3.class), is(not(nullValue())));
+                });
+            });
+		});
+	}
+
+	@Configuration
+	@AutoConfigurationPackage
+	@EnableAutoConfiguration
+	@EnableS3Stores(basePackageClasses=S3AutoConfigurationTest.class)
+	public static class TestConfigWithoutBeans {
+		// will be supplied by auto-configuration
+	}
+
+	public interface TestEntityRepository extends JpaRepository<TestEntity, Long> {
+	}
+
+	public interface TestEntityContentRepository extends ContentStore<TestEntity, String> {
+	}
+}

--- a/spring-content-autoconfigure/src/test/java/org/springframework/support/TestUtils.java
+++ b/spring-content-autoconfigure/src/test/java/org/springframework/support/TestUtils.java
@@ -1,0 +1,38 @@
+package org.springframework.support;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.Map;
+
+public final class TestUtils {
+
+    private TestUtils() {}
+
+    @SuppressWarnings("unchecked")
+    public static void setEnv(Map<String, String> newenv) throws Exception {
+        try {
+            Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
+            Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
+            theEnvironmentField.setAccessible(true);
+            Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
+            env.putAll(newenv);
+            Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField("theCaseInsensitiveEnvironment");
+            theCaseInsensitiveEnvironmentField.setAccessible(true);
+            Map<String, String> cienv = (Map<String, String>)theCaseInsensitiveEnvironmentField.get(null);
+            cienv.putAll(newenv);
+        } catch (NoSuchFieldException e) {
+            Class[] classes = Collections.class.getDeclaredClasses();
+            Map<String, String> env = System.getenv();
+            for(Class cl : classes) {
+                if("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
+                    Field field = cl.getDeclaredField("m");
+                    field.setAccessible(true);
+                    Object obj = field.get(env);
+                    Map<String, String> map = (Map<String, String>) obj;
+                    map.clear();
+                    map.putAll(newenv);
+                }
+            }
+        }
+    }
+}

